code/game/objects/items/melee/baton.dm

```diff
@@ -267,6 +270,16 @@
 		user.do_attack_animation(user)
 	return
 
+/obj/item/melee/baton/attack_self(user)
+	if(spriteSelected)
+		return //We already selected the sprite for the baton
+	var/spriteSelection = input(user, "Select a look for your baton (Classic_baton is wooden one, classic_baton_black is metalic nightstick)", "Baton Look") in list("classic_baton", "classic_baton_black")
```

Comment:
1. Make this use tgui_input
2. Make this use human readable names ("Classic baton" instead of "classic_baton"), then just have a dictionary to map to the icon states
3. Make this camelCase